### PR TITLE
Switch the repo to hardhat

### PR DESCRIPTION
**Pull request by AndriianChestnykh**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,54 +1,44 @@
...
+    "random-bytes": "^1.0.0"
+  },
+  "dependencies": {
+    "braces": ">=3.0.2",
+    "bs58": "^4.0.1",
+    "circomlib": "0.2.2",
+    "codecov": "^3.8.2",
+    "diff": ">=5.0.0",
+    "ethereumjs-util": "^7.1.4",
+    "ganache": "^7.0.2",
```
Comment (OBrezhniev) [inline package.json:34]: Is it still needed?

File: test/Iden3Helpers.test.js [hunk 2]
```diff
@@ -1,218 +1,188 @@
...
-/* global contract */
-/* global web3 */
-/* global assert */
-
 const ethUtil = require('ethereumjs-util');
-const mimcUnit = require("../node_modules/circomlib/src/mimc_gencontract.js");
-const HelpersTest = artifacts.require("../contracts/test/Iden3HelpersTest");
-const bs58 = require('bs58');
-const bigInt = require('big-integer');
+const mimcUnit = require("circomlib/src/mimc_gencontract.js");
```
Comment (OBrezhniev) [inline test/Iden3Helpers.test.js:2]: You can remove all mimc related code as it is not used.

File: package.json [hunk 3]
```diff
@@ -1,54 +1,44 @@
...
+    "async": "^3.2.3",
+    "big-integer": "^1.6.51",
+    "prettier": "^2.5.1",
+    "prettier-plugin-solidity": "^1.0.0-beta.19",
+    "random-bytes": "^1.0.0"
+  },
+  "dependencies": {
+    "braces": ">=3.0.2",
+    "bs58": "^4.0.1",
+    "circomlib": "0.2.2",
```
Comment (OBrezhniev) [inline package.json:30]: Probably makes sense to switch to newer circomlibjs.

File: scripts/sample-script.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  // If this script is run directly using `node` you may want to call compile
+  // manually to make sure everything is compiled
+  // await hre.run('compile');
+
+  // We get the contract to deploy
+  const Greeter = await hre.ethers.getContractFactory("Greeter");
+  const greeter = await Greeter.deploy("Hello, Hardhat!");
+
+  await greeter.deployed();
+
```
Comment (vmidyllic) [inline scripts/sample-script.js:21]: is a sample file needed?

File: test/Iden3Helpers.test.js (hunk 2 above)
Comment (AndriianChestnykh) [reply test/Iden3Helpers.test.js:2]: Which of the following we should keep then:
babyjubjub.test.js
eddsa-babyjubjub.test.js
Iden3Helpers.test.js

File: package.json (hunk 3 above)
Comment (AndriianChestnykh) [reply package.json:30]: Let's discuss which of the circomlib dependent tests we need to keep.

File: test/Iden3Helpers.test.js (hunk 2 above)
Comment (AndriianChestnykh) [reply test/Iden3Helpers.test.js:2]: Discussed and removed

File: package.json (hunk 3 above)
Comment (AndriianChestnykh) [reply package.json:30]: Discussed and decided to leave as is

File: scripts/sample-script.js (hunk 4 above)
Comment (AndriianChestnykh) [reply scripts/sample-script.js:21]: The script removed

File: README.md [hunk 5]
```diff
@@ -5,11 +5,3 @@
...
 
 **All code here is experimental and WIP**
-
-## Testing
-
-- Install truffle `npm i -g truffle`
-- Run the tests `truffle test` 
-- Run coverage tests `npm run coverage`
-- truffle migrate --network ropsten --skipDryRun
-- truffle migrate --network=development
```
[the remark this answers is not in the document]
Comment (AndriianChestnykh) [reply README.md]: done

**Review by OBrezhniev — APPROVED** [review]

**Pull request merged**